tip-router-operator-cli/scripts/setup-test-ledger.sh

```diff
@@ -20,20 +21,53 @@ create_keypair () {
   fi
 }
 
-create_vote_accounts () {
+# Function to create keypairs and serialize accounts
+prepare_keypairs_and_serialize() {
   max_validators=$1
   validator_file=$2
-  for number in $(seq 1 "$max_validators")
-  do
+
+  for number in $(seq 1 "$max_validators"); do
+    # Create keypairs for identity, vote, and withdrawer
     create_keypair "$keys_dir/identity_$number.json"
     create_keypair "$keys_dir/vote_$number.json"
     create_keypair "$keys_dir/withdrawer_$number.json"
-    solana create-vote-account "$keys_dir/vote_$number.json" "$keys_dir/identity_$number.json" "$keys_dir/withdrawer_$number.json" --commission 1
+
+    # Get the public key of the vote account
     vote_pubkey=$(solana-keygen pubkey "$keys_dir/vote_$number.json")
+
+    # Extract the public key from the identity keypair
+    merkle_root_upload_authority=$(solana-keygen pubkey "$keys_dir/identity_$number.json")
+
+    # Append the vote public key to the validator file
     echo "$vote_pubkey" >> "$validator_file"
+
+    # Dynamically run the Rust script with the vote_pubkey
+    RUST_LOG=info cargo run --bin serialize-accounts -- \
+      --validator-vote-account "$vote_pubkey" \
+      --merkle-root-upload-authority "$merkle_root_upload_authority" \
+      --epoch-created-at 4 \
+      --validator-commission-bps 1 \
+      --expires-at 1000 \
+      --bump 1
   done
 }
 
+# Function to create vote accounts
+create_vote_accounts() {
+  max_validators=$1
+  validator_file=$2
+
+  for number in $(seq 1 "$max_validators"); do
+    # Create the vote account
+    solana create-vote-account \
+      "$keys_dir/vote_$number.json" \
+      "$keys_dir/identity_$number.json" \
+      "$keys_dir/withdrawer_$number.json" \
+      --commission 1
+  done
+}
+
+
```

Comment:
is this still needed with `prepare_keypairs_and_serialize`?

Reply:
yeah, that's creating the vote accounts